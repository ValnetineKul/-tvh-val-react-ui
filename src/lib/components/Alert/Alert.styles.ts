import { makeStyles } from '../../../themes/core';

import type { AlertProps } from './Alert.types';

type RestrictedMuiAlertProps = Pick<AlertProps, 'status'>;
type Props = RestrictedMuiAlertProps & {
  shouldAddAutoLeftMarginToFirstDescendent: boolean;
};

const useStyles = makeStyles<Props, 'icon'>({ name: 'Alert' })((theme, props, classes) => ({
  root: {
    display: 'flex',
    alignItems: 'flex-start',
    backgroundColor: 'inherit',
    padding: `${theme.layoutSpacing['spacing/300']} ${theme.layoutSpacing['spacing/400']}`,
    borderRadius: theme.borderRadius['borderRadius/none'],
    [`& .${classes.icon}`]: {
      color: `${theme.color[`icon/${props.status}` as const]}`,
    },

    [theme.breakpoints.up('sm')]: {
      padding: theme.layoutSpacing['spacing/300'],
      justifyContent: 'center',
      '& > *:first-of-type': {
        marginLeft: props.shouldAddAutoLeftMarginToFirstDescendent ? 'auto' : undefined,
      },
    },
  },
  icon: {
    padding: `${theme.layoutSpacing['spacing/200']} ${theme.layoutSpacing['spacing/000']}`,
    marginRight: theme.layoutSpacing['spacing/300'],
  },
  message: {
    color: theme.color['text/onSurface100/default'],
    padding: `${theme.layoutSpacing['spacing/200']} ${theme.layoutSpacing['spacing/000']}`,
    textAlign: 'left',
  },
  action: {
    padding: `${theme.layoutSpacing['spacing/000']} ${theme.layoutSpacing['spacing/000']} ${theme.layoutSpacing['spacing/000']} ${theme.layoutSpacing['spacing/300']}`,
    marginRight: theme.layoutSpacing['spacing/000'],
  },
}));

export default useStyles;
