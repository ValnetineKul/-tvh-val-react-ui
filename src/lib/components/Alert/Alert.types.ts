import type { ReactNode } from 'react';

import type { AlertProps as MuiAlertProps } from '@mui/material';

import type { DataAttributes } from '../../types/common';

export const Status = {
  info: 'info',
  warning: 'warning',
  error: 'error',
  success: 'success',
} as const;

export type StatusType = typeof Status[keyof typeof Status];

type RestrictedMuiAlertProps = Pick<MuiAlertProps, 'icon'>;

export interface AlertProps extends RestrictedMuiAlertProps {
  status: StatusType;
  onClose?: () => void;
  message: ReactNode;
  action?: ReactNode;
  className?: string;
  classesOverrides?: MuiAlertProps['classes'];
  containerClassName?: string;
  surfaceClassName?: string;
  closable?: boolean;
  closeButtonProps?: DataAttributes;
}
