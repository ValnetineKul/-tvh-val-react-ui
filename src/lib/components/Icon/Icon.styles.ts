import { makeStyles } from '../../../themes/core';

const useStyles = makeStyles({ name: 'Icon' })(() => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  sizeXl: {
    fontSize: '64px !important',
    width: '80px',
    height: '80px',
    '& svg': {
      width: '80px',
      height: '80px',
    },
  },
  sizeMd: {
    fontSize: '19px !important',
    width: '24px',
    height: '24px',
    '& svg': {
      width: '24px',
      height: '24px',
    },
  },
  sizeSm: {
    fontSize: '12px !important',
    width: '16px',
    height: '16px',
    '& svg': {
      width: '16px',
      height: '16px',
    },
  },
}));

export default useStyles;
