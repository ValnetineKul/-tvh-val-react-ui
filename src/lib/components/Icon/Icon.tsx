import React from 'react';
import SvgIcon from '@mui/material/SvgIcon';
import { capitalize } from '@mui/material/utils';
import useStyles from './Icon.styles';

export interface IconProps extends React.AriaAttributes {
  icon: React.FunctionComponent;
  size?: 'md' | 'sm' | 'xl';
  className?: string;
  title?: string;
  isMultiColor?: boolean;
}

const Icon = ({ icon, size = 'md', className, title, isMultiColor = false, ...props }: IconProps) => {
  const { classes, cx } = useStyles();
  const sizeClassName = (classes as Record<string, string>)[`size${capitalize(size)}`];

  const IconComponent = typeof icon === 'string' ? React.lazy(() => import(`./icons/${icon}`)) : icon;

  return (
    <div className={cx(classes.root, sizeClassName, className)}>
      <SvgIcon
        htmlColor={isMultiColor ? 'transparent' : 'inherit'}
        component={IconComponent as React.ElementType}
        // a11y mui issue: combination "component + titleAccess" does not work: https://github.com/mui/material-ui/issues/33421
        titleAccess={title}
        {...props}
      />
    </div>
  );
};

export default Icon;
