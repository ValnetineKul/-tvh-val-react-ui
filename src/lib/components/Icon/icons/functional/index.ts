export { ReactComponent as AddressBook } from './svg/addressBook.svg';
export { ReactComponent as Analytics } from './svg/analytics.svg';
export { ReactComponent as AngleDown } from './svg/angleDown.svg';
export { ReactComponent as AngleLeft } from './svg/angleLeft.svg';
export { ReactComponent as AngleRight } from './svg/angleRight.svg';
export { ReactComponent as AngleUp } from './svg/angleUp.svg';
export { ReactComponent as ArrowAltCircleLeft } from './svg/arrowAltCircleLeft.svg';
export { ReactComponent as ArrowDown } from './svg/arrowDown.svg';
export { ReactComponent as ArrowDownArrowUp } from './svg/arrowDownArrowUp.svg';
export { ReactComponent as ArrowLeft } from './svg/arrowLeft.svg';
export { ReactComponent as ArrowLeftToLine } from './svg/arrowLeftToLine.svg';
export { ReactComponent as ArrowRight } from './svg/arrowRight.svg';
export { ReactComponent as ArrowRightArrowLeft } from './svg/arrowRightArrowLeft.svg';
export { ReactComponent as ArrowRightToLine } from './svg/arrowRightToLine.svg';
export { ReactComponent as ArrowUp } from './svg/arrowUp.svg';
export { ReactComponent as ArrowUpRightFromSquare } from './svg/arrowUpRightFromSquare.svg';
export { ReactComponent as Atlas } from './svg/atlas.svg';
export { ReactComponent as Award } from './svg/award.svg';
export { ReactComponent as Badge } from './svg/badge.svg';
export { ReactComponent as BadgePercent } from './svg/badgePercent.svg';
export { ReactComponent as BallotCheck } from './svg/ballotCheck.svg';
export { ReactComponent as Bars } from './svg/bars.svg';
export { ReactComponent as BatteryFull } from './svg/batteryFull.svg';
export { ReactComponent as BatteryHalf } from './svg/batteryHalf.svg';
export { ReactComponent as BatteryLow } from './svg/batteryLow.svg';
export { ReactComponent as Bell } from './svg/bell.svg';
export { ReactComponent as Book } from './svg/book.svg';
export { ReactComponent as BookAlt } from './svg/bookAlt.svg';
export { ReactComponent as BookmarkCheck } from './svg/bookmarkCheck.svg';
export { ReactComponent as BookOpen } from './svg/bookOpen.svg';
export { ReactComponent as Box } from './svg/box.svg';
export { ReactComponent as BoxAlt } from './svg/boxAlt.svg';
export { ReactComponent as Boxes } from './svg/boxes.svg';
export { ReactComponent as BoxFull } from './svg/boxFull.svg';
export { ReactComponent as BoxHeart } from './svg/boxHeart.svg';
export { ReactComponent as BoxOpen } from './svg/boxOpen.svg';
export { ReactComponent as Briefcase } from './svg/briefcase.svg';
export { ReactComponent as Building } from './svg/building.svg';
export { ReactComponent as BulkyItem } from './svg/bulkyItem.svg';
export { ReactComponent as BullseyeArrow } from './svg/bullseyeArrow.svg';
export { ReactComponent as BullseyePointer } from './svg/bullseyePointer.svg';
export { ReactComponent as Calculator } from './svg/calculator.svg';
export { ReactComponent as Calendar } from './svg/calendar.svg';
export { ReactComponent as CalendarCheck } from './svg/calendarCheck.svg';
export { ReactComponent as CalendarStar } from './svg/calendarStar.svg';
export { ReactComponent as Camera } from './svg/camera.svg';
export { ReactComponent as CarMechanic } from './svg/carMechanic.svg';
export { ReactComponent as CartArrowDown } from './svg/cartArrowDown.svg';
export { ReactComponent as Cat } from './svg/cat.svg';
export { ReactComponent as ChartLine } from './svg/chartLine.svg';
export { ReactComponent as ChartPie } from './svg/chartPie.svg';
export { ReactComponent as Check } from './svg/check.svg';
export { ReactComponent as CheckCircle } from './svg/checkCircle.svg';
export { ReactComponent as ChessKnight } from './svg/chessKnight.svg';
export { ReactComponent as ClipboardListCheck } from './svg/clipboardListCheck.svg';
export { ReactComponent as Clock } from './svg/clock.svg';
export { ReactComponent as Cog } from './svg/cog.svg';
export { ReactComponent as Cogs } from './svg/cogs.svg';
export { ReactComponent as Coin } from './svg/coin.svg';
export { ReactComponent as Coins } from './svg/coins.svg';
export { ReactComponent as Comment } from './svg/comment.svg';
export { ReactComponent as CommentsAlt } from './svg/commentsAlt.svg';
export { ReactComponent as ContainerStorage } from './svg/containerStorage.svg';
export { ReactComponent as ConveyorBelt } from './svg/conveyorBelt.svg';
export { ReactComponent as ConveyorBeltSlash } from './svg/conveyorBeltSlash.svg';
export { ReactComponent as Copy } from './svg/copy.svg';
export { ReactComponent as CreditCard } from './svg/creditCard.svg';
export { ReactComponent as DesktopAlt } from './svg/desktopAlt.svg';
export { ReactComponent as Dolly } from './svg/dolly.svg';
export { ReactComponent as DollyFlatbed } from './svg/dollyFlatbed.svg';
export { ReactComponent as DollyFlatbedAlt } from './svg/dollyFlatbedAlt.svg';
export { ReactComponent as Download } from './svg/download.svg';
export { ReactComponent as EllipsisH } from './svg/ellipsisH.svg';
export { ReactComponent as Envelope } from './svg/envelope.svg';
export { ReactComponent as ExclamationCircle } from './svg/exclamationCircle.svg';
export { ReactComponent as ExclamationTriangle } from './svg/exclamationTriangle.svg';
export { ReactComponent as Eye } from './svg/eye.svg';
export { ReactComponent as EyeSlash } from './svg/eyeSlash.svg';
export { ReactComponent as Fast } from './svg/fast.svg';
export { ReactComponent as FileAlt } from './svg/fileAlt.svg';
export { ReactComponent as FileCheck } from './svg/fileCheck.svg';
export { ReactComponent as FileDownload } from './svg/fileDownload.svg';
export { ReactComponent as FileInvoice } from './svg/fileInvoice.svg';
export { ReactComponent as FileUpload } from './svg/fileUpload.svg';
export { ReactComponent as Filter } from './svg/filter.svg';
export { ReactComponent as FlagAlt } from './svg/flagAlt.svg';
export { ReactComponent as Folder } from './svg/folder.svg';
export { ReactComponent as Forklift } from './svg/forklift.svg';
export { ReactComponent as Fragile } from './svg/fragile.svg';
export { ReactComponent as Gauge } from './svg/gauge.svg';
export { ReactComponent as GaugeMax } from './svg/gaugeMax.svg';
export { ReactComponent as GaugeMin } from './svg/gaugeMin.svg';
export { ReactComponent as Gift } from './svg/gift.svg';
export { ReactComponent as GiftCard } from './svg/giftCard.svg';
export { ReactComponent as Gifts } from './svg/gifts.svg';
export { ReactComponent as Globe } from './svg/globe.svg';
export { ReactComponent as GlobeAfrica } from './svg/globeAfrica.svg';
export { ReactComponent as GlobeAmericas } from './svg/globeAmericas.svg';
export { ReactComponent as GlobeAsia } from './svg/globeAsia.svg';
export { ReactComponent as GlobeEurope } from './svg/globeEurope.svg';
export { ReactComponent as GraduationCap } from './svg/graduationCap.svg';
export { ReactComponent as Grid } from './svg/grid.svg';
export { ReactComponent as HandHoldingBox } from './svg/handHoldingBox.svg';
export { ReactComponent as HandPointerSwipe } from './svg/handPointerSwipe.svg';
export { ReactComponent as Handshake } from './svg/handshake.svg';
export { ReactComponent as Headset } from './svg/headset.svg';
export { ReactComponent as HelmetSafety } from './svg/helmetSafety.svg';
export { ReactComponent as Hiking } from './svg/hiking.svg';
export { ReactComponent as History } from './svg/history.svg';
export { ReactComponent as Home } from './svg/home.svg';
export { ReactComponent as HorizontalRule } from './svg/horizontalRule.svg';
export { ReactComponent as Image } from './svg/image.svg';
export { ReactComponent as Inbox } from './svg/inbox.svg';
export { ReactComponent as Indent } from './svg/indent.svg';
export { ReactComponent as Infinite } from './svg/infinite.svg';
export { ReactComponent as InfoCircle } from './svg/infoCircle.svg';
export { ReactComponent as Inventory } from './svg/inventory.svg';
export { ReactComponent as InvoiceOverdue } from './svg/invoiceOverdue.svg';
export { ReactComponent as Language } from './svg/language.svg';
export { ReactComponent as Leaf } from './svg/leaf.svg';
export { ReactComponent as LevelRight } from './svg/levelRight.svg';
export { ReactComponent as LightbulbOn } from './svg/lightbulbOn.svg';
export { ReactComponent as Link } from './svg/link.svg';
export { ReactComponent as LinkSlash } from './svg/linkSlash.svg';
export { ReactComponent as List } from './svg/list.svg';
export { ReactComponent as Lock } from './svg/lock.svg';
export { ReactComponent as LockOpen } from './svg/lockOpen.svg';
export { ReactComponent as MapLocationDot } from './svg/mapLocationDot.svg';
export { ReactComponent as MapMarkerAlt } from './svg/mapMarkerAlt.svg';
export { ReactComponent as Medal } from './svg/medal.svg';
export { ReactComponent as Megaphone } from './svg/megaphone.svg';
export { ReactComponent as Microchip } from './svg/microchip.svg';
export { ReactComponent as Minus } from './svg/minus.svg';
export { ReactComponent as MobileButton } from './svg/mobileButton.svg';
export { ReactComponent as MoneyBillWave } from './svg/moneyBillWave.svg';
export { ReactComponent as MoneyCheckEdit } from './svg/moneyCheckEdit.svg';
export { ReactComponent as Narwhal } from './svg/narwhal.svg';
export { ReactComponent as NonReturnableItem } from './svg/nonReturnableItem.svg';
export { ReactComponent as Outdent } from './svg/outdent.svg';
export { ReactComponent as Pallet } from './svg/pallet.svg';
export { ReactComponent as Paperclip } from './svg/paperclip.svg';
export { ReactComponent as PaperPlane } from './svg/paperPlane.svg';
export { ReactComponent as ParachuteBox } from './svg/parachuteBox.svg';
export { ReactComponent as Pen } from './svg/pen.svg';
export { ReactComponent as PeopleArrowsLeftRight } from './svg/peopleArrowsLeftRight.svg';
export { ReactComponent as PeopleGroup } from './svg/peopleGroup.svg';
export { ReactComponent as PersonDolly } from './svg/personDolly.svg';
export { ReactComponent as PersonRays } from './svg/personRays.svg';
export { ReactComponent as Phone } from './svg/phone.svg';
export { ReactComponent as Plane } from './svg/plane.svg';
export { ReactComponent as PlaneSlash } from './svg/planeSlash.svg';
export { ReactComponent as Play } from './svg/play.svg';
export { ReactComponent as PlayCircle } from './svg/playCircle.svg';
export { ReactComponent as Plus } from './svg/plus.svg';
export { ReactComponent as Print } from './svg/print.svg';
export { ReactComponent as PuzzlePiece } from './svg/puzzlePiece.svg';
export { ReactComponent as QrCode } from './svg/qrCode.svg';
export { ReactComponent as QuestionCircle } from './svg/questionCircle.svg';
export { ReactComponent as Random } from './svg/random.svg';
export { ReactComponent as Recycle } from './svg/recycle.svg';
export { ReactComponent as Redo } from './svg/redo.svg';
export { ReactComponent as ResearchRequest } from './svg/researchRequest.svg';
export { ReactComponent as ReturnableItem } from './svg/returnableItem.svg';
export { ReactComponent as Ribbon } from './svg/ribbon.svg';
export { ReactComponent as RocketLaunch } from './svg/rocketLaunch.svg';
export { ReactComponent as Route } from './svg/route.svg';
export { ReactComponent as RulerHorizontal } from './svg/rulerHorizontal.svg';
export { ReactComponent as Scanner } from './svg/scanner.svg';
export { ReactComponent as Screwdriver } from './svg/screwdriver.svg';
export { ReactComponent as Search } from './svg/search.svg';
export { ReactComponent as SearchMinus } from './svg/searchMinus.svg';
export { ReactComponent as SearchPlus } from './svg/searchPlus.svg';
export { ReactComponent as ShieldCheck } from './svg/shieldCheck.svg';
export { ReactComponent as Ship } from './svg/ship.svg';
export { ReactComponent as ShippingFast } from './svg/shippingFast.svg';
export { ReactComponent as ShippingTimed } from './svg/shippingTimed.svg';
export { ReactComponent as ShoppingBasket } from './svg/shoppingBasket.svg';
export { ReactComponent as ShoppingCart } from './svg/shoppingCart.svg';
export { ReactComponent as SignOut } from './svg/signOut.svg';
export { ReactComponent as Smile } from './svg/smile.svg';
export { ReactComponent as Snowplow } from './svg/snowplow.svg';
export { ReactComponent as SolarPanel } from './svg/solarPanel.svg';
export { ReactComponent as Sparkles } from './svg/sparkles.svg';
export { ReactComponent as Star } from './svg/star.svg';
export { ReactComponent as Stopwatch } from './svg/stopwatch.svg';
export { ReactComponent as Sync } from './svg/sync.svg';
export { ReactComponent as Tag } from './svg/tag.svg';
export { ReactComponent as Tags } from './svg/tags.svg';
export { ReactComponent as ThumbsUp } from './svg/thumbsUp.svg';
export { ReactComponent as Ticket } from './svg/ticket.svg';
export { ReactComponent as TimelineArrow } from './svg/timelineArrow.svg';
export { ReactComponent as Times } from './svg/times.svg';
export { ReactComponent as TimesCircle } from './svg/timesCircle.svg';
export { ReactComponent as Tools } from './svg/tools.svg';
export { ReactComponent as Tractor } from './svg/tractor.svg';
export { ReactComponent as Trash } from './svg/trash.svg';
export { ReactComponent as Trophy } from './svg/trophy.svg';
export { ReactComponent as TrophyAlt } from './svg/trophyAlt.svg';
export { ReactComponent as Truck } from './svg/truck.svg';
export { ReactComponent as TruckContainer } from './svg/truckContainer.svg';
export { ReactComponent as TruckFree } from './svg/truckFree.svg';
export { ReactComponent as TruckFreeSlash } from './svg/truckFreeSlash.svg';
export { ReactComponent as TruckLoading } from './svg/truckLoading.svg';
export { ReactComponent as TruckPickup } from './svg/truckPickup.svg';
export { ReactComponent as TruckPlow } from './svg/truckPlow.svg';
export { ReactComponent as UnknownPart } from './svg/unknownPart.svg';
export { ReactComponent as Unlink } from './svg/unlink.svg';
export { ReactComponent as UpRightAndDownLeftFromCenter } from './svg/upRightAndDownLeftFromCenter.svg';
export { ReactComponent as User } from './svg/user.svg';
export { ReactComponent as UserCircle } from './svg/userCircle.svg';
export { ReactComponent as UserHeadset } from './svg/userHeadset.svg';
export { ReactComponent as Users } from './svg/users.svg';
export { ReactComponent as UserShield } from './svg/userShield.svg';
export { ReactComponent as Video } from './svg/video.svg';
export { ReactComponent as VideoSlash } from './svg/videoSlash.svg';
export { ReactComponent as View360 } from './svg/view360.svg';
export { ReactComponent as Wallet } from './svg/wallet.svg';
export { ReactComponent as WalletCheck } from './svg/walletCheck.svg';
export { ReactComponent as WarehouseFull } from './svg/warehouseFull.svg';
export { ReactComponent as WeightHanging } from './svg/weightHanging.svg';
export { ReactComponent as Wheat } from './svg/wheat.svg';
export { ReactComponent as Wifi } from './svg/wifi.svg';
export { ReactComponent as WifiSlash } from './svg/wifiSlash.svg';
export { ReactComponent as Wrench } from './svg/wrench.svg';
