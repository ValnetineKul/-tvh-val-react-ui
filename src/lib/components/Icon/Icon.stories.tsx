import type { FunctionComponent, SVGProps } from 'react';
import React from 'react';
import Typography from '../Typography';
import { createMeta } from '../../story-utils';
import Icon from './Icon';
import * as countryIcons from './icons/country';
import * as functionalIcons from './icons/functional';
import * as productCategory from './icons/productCategory';
import * as machinery from './icons/machinery';
import * as commercial from './icons/commercial';
import * as brand from './icons/brand';

type Size = 'xl' | 'md' | 'sm';

const renderIconsListWithLetter = <
  T extends Record<string, FunctionComponent<SVGProps<SVGSVGElement> & { title?: string | undefined }>>
>(
  icons: T,
  size: Size,
  isMultiColor = false
) => {
  const iconsNames: string[] = Object.keys(icons);

  iconsNames.sort((a: string, b: string) => a.localeCompare(b));

  let letter = '';
  return iconsNames.map((name) => {
    if (name[0] === letter) {
      return null;
    }
    [letter] = name;

    return (
      <div
        style={{
          display: 'flex',
          flexWrap: 'wrap',
          alignItems: 'center',
          padding: 10,
          marginBottom: 20,
          border: '1px solid black',
          borderRadius: 12,
        }}
        key={name}
      >
        <div style={{ marginRight: 20 }}>
          <Typography variant="h1" headerType="commercial">
            {name[0]}
          </Typography>
        </div>

        {iconsNames.map((iconName) => {
          if (iconName[0] !== letter) {
            return null;
          }

          return (
            <div style={{ padding: 10, display: 'flex', flexDirection: 'column', alignItems: 'center' }} key={iconName}>
              <Icon isMultiColor={isMultiColor} icon={icons[iconName as keyof T]} size={size} />

              <Typography>{iconName}</Typography>
            </div>
          );
        })}
      </div>
    );
  });
};

const FunctionalIconsListTemplate = ({ size }: { size: Size }) => {
  return (
    <div style={{ display: 'flex', flexDirection: 'column' }}>{renderIconsListWithLetter(functionalIcons, size)}</div>
  );
};

export const FunctionalIconsList = FunctionalIconsListTemplate.bind({});
FunctionalIconsList.args = {
  size: 'md',
};
FunctionalIconsList.storyName = 'Functional icons list';

const CountryIconsListTemplate = ({ size }: { size: Size }) => {
  return (
    <div style={{ display: 'flex', flexDirection: 'column' }}>
      {renderIconsListWithLetter(countryIcons, size, true)}
    </div>
  );
};

export const CountryIconsList = CountryIconsListTemplate.bind({});
CountryIconsList.args = {
  size: 'md',
};
CountryIconsList.storyName = 'Country icons list';

const ProductCategoryIconsListTemplate = ({ size }: { size: Size }) => {
  return (
    <div style={{ display: 'flex', flexDirection: 'column' }}>{renderIconsListWithLetter(productCategory, size)}</div>
  );
};

export const ProductCategoryIconsList = ProductCategoryIconsListTemplate.bind({});
ProductCategoryIconsList.args = {
  size: 'md',
};
ProductCategoryIconsList.storyName = 'Product category icons list';

const MachineryIconsListTemplate = ({ size }: { size: Size }) => {
  return <div style={{ display: 'flex', flexDirection: 'column' }}>{renderIconsListWithLetter(machinery, size)}</div>;
};

export const MachineryIconsList = MachineryIconsListTemplate.bind({});
MachineryIconsList.args = {
  size: 'md',
};
MachineryIconsList.storyName = 'Machinery icons list';

const CommercialIconsListTemplate = ({ size }: { size: Size }) => {
  return <div style={{ display: 'flex', flexDirection: 'column' }}>{renderIconsListWithLetter(commercial, size)}</div>;
};

export const CommercialIconsList = CommercialIconsListTemplate.bind({});
CommercialIconsList.args = {
  size: 'md',
};
CommercialIconsList.storyName = 'Commercial icons list';

const BrandIconsListTemplate = ({ size }: { size: Size }) => {
  return <div style={{ display: 'flex', flexDirection: 'column' }}>{renderIconsListWithLetter(brand, size)}</div>;
};

export const BrandIconsList = BrandIconsListTemplate.bind({});
BrandIconsList.args = {
  size: 'md',
};
BrandIconsList.storyName = 'Brand icons list';

export default createMeta({
  component: Icon,
  title: 'Foundations/Icon',
  argTypes: {
    icon: {
      control: {
        disable: true,
      },
      table: {
        type: { summary: 'FontAwesome icon definition' },
      },
    },
    isMultiColor: {
      control: {
        disable: true,
      },
    },
    title: {
      control: {
        disable: true,
      },
    },
  },
});
