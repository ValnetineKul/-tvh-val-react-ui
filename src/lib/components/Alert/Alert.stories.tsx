import React from 'react';

import { action } from '@storybook/addon-actions';
import type { ComponentStory, ComponentMeta } from '@storybook/react';
import Alert from './Alert';
import InlineMessage from '../InlineMessage';
import { Narwhal, WifiSlash, Wrench } from '../Icon/icons/functional';
import Icon from '../Icon';
import Typography from '../Typography';
import { Status } from './Alert.types';
import Button from '../Buttons/Button';
import { makeStyles } from '../../../themes/core';

const useStyles = makeStyles()(() => ({
  link: {
    verticalAlign: 'baseline',
  },
}));

export default {
  title: 'Components/Alerts/Alert',
  component: Alert,
  args: {
    status: 'info',
    message: 'example alert message...',
  },
  argTypes: {
    status: {
      options: [...Object.values(Status)],
      control: {
        type: 'select',
      },
    },
  },
  parameters: {
    layout: 'fullscreen',
  },
} as ComponentMeta<typeof Alert>;

const Template: ComponentStory<typeof Alert> = (args) => <Alert {...args} />;

export const Primary = Template.bind({});

export const WithInlineMessage = Template.bind({});
WithInlineMessage.args = {
  onClose: action('close'),
  message: <InlineMessage message="Inline message." actionLabel="Action" onAction={action('onAction')} />,
};

export const Custom1 = Template.bind({});
Custom1.args = {
  status: 'warning',
  icon: <Icon icon={WifiSlash} />,
  message: (
    <InlineMessage message="You are in offline mode." actionLabel="More Information" onAction={action('onAction')} />
  ),
};

export const Custom2 = Template.bind({});
Custom2.args = {
  icon: <Icon icon={Wrench} />,
  message: (
    <Typography>
      Maintenance is scheduled on{' '}
      <Typography component="b" weight="emphasis">
        Saturday, December 10 at 16:00 UTC.
      </Typography>{' '}
      The platform will be unavailable for a couple of hours.
    </Typography>
  ),
};

const TemplateCustom3: ComponentStory<typeof Alert> = (args) => {
  const { classes } = useStyles();

  return (
    <Alert
      {...args}
      message={
        <Typography>
          This is an alert message with a <Button label="link" variant="link" href="/" className={classes.link} />,{' '}
          <Typography component="b" weight="emphasis">
            bold text
          </Typography>{' '}
          and a custom icon
        </Typography>
      }
    />
  );
};

export const Custom3 = TemplateCustom3.bind({});
Custom3.args = {
  icon: <Icon icon={Narwhal} />,
};
