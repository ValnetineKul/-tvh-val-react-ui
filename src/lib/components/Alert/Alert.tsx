import * as React from 'react';
import { default as MuiAlert } from '@mui/material/Alert';
import { capitalize } from '@mui/material/utils';

import { CheckCircle, ExclamationCircle, ExclamationTriangle, InfoCircle, Times } from '../Icon/icons/functional';
import Icon from '../Icon';
import IconButton from '../Buttons/IconButton';

import Typography from '../Typography';

import useStyles from './Alert.styles';
import type { AlertProps } from './Alert.types';
import Surface from '../Surfaces/Surface';
import Collapse from '../Collapse';

const iconMapping = {
  success: <Icon icon={CheckCircle} />,
  warning: <Icon icon={ExclamationTriangle} />,
  error: <Icon icon={ExclamationCircle} />,
  info: <Icon icon={InfoCircle} />,
} as const;

const isNull = (value: unknown): value is null => value === null;
const isUndefined = (value: unknown): value is undefined => typeof value === 'undefined';
const isDefined = (value: unknown): boolean => !isNull(value) && !isUndefined(value);

const Alert: React.FC<AlertProps> = ({
  className,
  classesOverrides,
  icon,
  status,
  message,
  action,
  closable,
  onClose,
  closeButtonProps,
  containerClassName,
  surfaceClassName,
}) => {
  const [isClosed, setIsClosed] = React.useState(false);

  const hasAction = closable || isDefined(action);
  const { cx, classes } = useStyles({
    status,
    shouldAddAutoLeftMarginToFirstDescendent: hasAction,
  });

  const handleClose = () => {
    setIsClosed(true);
    onClose?.();
  };

  const renderAction = () => {
    return (
      <>
        {action}
        {closable && (
          <IconButton
            {...closeButtonProps}
            aria-label="close"
            size="md"
            onClick={handleClose}
            icon={<Icon icon={Times} />}
          />
        )}
      </>
    );
  };

  const renderMessage = () => {
    if (typeof message === 'string') {
      return (
        <Typography component="div" variant="body500">
          {message}
        </Typography>
      );
    }

    return message;
  };

  return (
    <Collapse open={!isClosed} className={containerClassName}>
      <Surface color={capitalize(status)} className={surfaceClassName}>
        <MuiAlert
          className={cx(classes.root, className)}
          classes={{
            icon: cx(classes.icon, classesOverrides?.icon),
            message: cx(classes.message, classesOverrides?.message),
            action: cx(classes.action, classesOverrides?.action),
          }}
          severity={status}
          iconMapping={iconMapping}
          icon={icon}
          action={hasAction ? renderAction() : undefined}
        >
          {renderMessage()}
        </MuiAlert>
      </Surface>
    </Collapse>
  );
};

export default Alert;
