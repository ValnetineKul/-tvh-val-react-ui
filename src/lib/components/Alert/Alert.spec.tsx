import React from 'react';
import { render, screen, userEvent } from '../../test-utils';
import { Narwhal } from '../Icon/icons/functional';
import Icon from '../Icon';
import Alert from './Alert';
import type { AlertProps } from './Alert.types';

const setup = (props?: Partial<AlertProps>) => {
  return render(<Alert status="info" message="an example alert message" {...props} />);
};

describe('Alert', () => {
  it('should render correctly', () => {
    const { container } = setup();
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByText(/an example alert message/i)).toBeInTheDocument();
  });

  it('should render a close button', () => {
    const onCloseMock = jest.fn();
    setup({ closable: true, onClose: onCloseMock });

    const closeButton = screen.getByRole('button', { name: /close/i });
    expect(closeButton).toBeInTheDocument();

    userEvent.click(closeButton);
    expect(onCloseMock).toHaveBeenCalled();
  });

  describe('should set the correct icon', () => {
    const cases = [
      ['info', 'infoCircle.svg'],
      ['warning', 'exclamationTriangle.svg'],
      ['success', 'checkCircle.svg'],
      ['error', 'exclamationCircle.svg'],
    ];
    test.each(cases)('given %p, returns %p', (input: AlertProps['status'], expected: string) => {
      setup({ status: input });
      expect(screen.getByText(expected, { selector: 'svg' })).toBeInTheDocument();
    });
    test('custom icon', () => {
      setup({ icon: <Icon icon={Narwhal} /> });
      expect(screen.getByText('narwhal.svg', { selector: 'svg' })).toBeInTheDocument();
    });
  });
});
