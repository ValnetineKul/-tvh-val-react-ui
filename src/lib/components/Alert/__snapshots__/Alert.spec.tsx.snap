// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render correctly 1`] = `
<div
  class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered mui-pwcg7p-MuiCollapse-root"
  style="min-height: 0px;"
>
  <div
    class="MuiCollapse-wrapper MuiCollapse-vertical mui-smkl36-MuiCollapse-wrapper"
  >
    <div
      class="MuiCollapse-wrapperInner MuiCollapse-vertical mui-9l5vo-MuiCollapse-wrapperInner"
    >
      <div
        class="tss-i0v2ei-Surface-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard tss-8rfudm-Alert-root mui-ig7n8t-MuiPaper-root-MuiAlert-root"
          role="alert"
        >
          <div
            class="tss-13p8oay-Alert-icon tss-5-Alert-icon-ref MuiAlert-icon mui-1ytlwq5-MuiAlert-icon"
          >
            <div
              class="tss-w1sjks-Icon-root-Icon-sizeMd"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-i4bv87-MuiSvgIcon-root"
                color="inherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                infoCircle.svg
              </svg>
            </div>
          </div>
          <div
            class="tss-1c1sssm-Alert-message MuiAlert-message mui-acap47-MuiAlert-message"
          >
            <div
              class="tss-3-Typography-weightRegular-ref tss-1n3cm1e-Typography-root-Typography-weightRegular-Typography-body500"
            >
              an example alert message
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
